bot/gif_command.go

```diff
@@ -3,39 +3,37 @@ package bot
 import (
 	"fmt"
 	"io"
-	"math/rand"
 	"net/url"
 	"strings"
 
 	"github.com/urfave/cli"
 	"github.com/zpatrick/rclient"
 )
 
-// Endpoint for the gif api
-const GiphyAPIEndpoint = "https://api.giphy.com"
+// TenorAPIEndpoint for the gif api
+const TenorAPIEndpoint = "https://api.tenor.com/"
 
-// GiphySearchResponse is the response type for a Giphy API search
-type GiphySearchResponse struct {
-	Gifs []GiphyGif `json:"data"`
+// TenorSearchResponse is the response type for a Tenor API search
+type TenorSearchResponse struct {
+	Gifs []Gif `json:"results"`
 }
 
-// GiphyGif holds information about a Gif from Giphy
-type GiphyGif struct {
-	URL string `json:"bitly_gif_url"`
+// Gif holds information about a Gif from Tenor
+type Gif struct {
+	URL string `json:"itemurl"`
 }
 
 // NewGIFCommand returns a cli.Command that manages !gif
-func NewGIFCommand(endpoint, token string, w io.Writer) cli.Command {
+func NewGIFCommand(endpoint, key string, w io.Writer) cli.Command {
 	client := rclient.NewRestClient(endpoint)
 	return cli.Command{
 		Name:      "!gif",
 		Usage:     "display a gif",
 		ArgsUsage: "args...",
 		Flags: []cli.Flag{
-			cli.StringFlag{
-				Name:  "rating",
-				Value: "pg-13",
-				Usage: "the MPAA-style rating for the search, e.g. 'pg', 'pg-13', 'r'",
+			cli.BoolFlag{
+				Name:  "explicit",
+				Usage: "This will turn off safe search https://tenor.com/gifapi/documentation#safesearch",
```

Comment:
[nit] let's try and keep flag descriptions short & sweet. Maybe something just like "enable explicit content"?